connectors/source.py

```diff
@@ -107,6 +132,114 @@ def is_empty(self):
     def to_dict(self):
         return dict(self._raw_config)
 
+    def is_valid(self):
+        """Validates every Field against its `validations`.
+
+        Raises ConfigurableFieldValueError if any validation errors are found,
+        otherwise returns True.
+        """
+        validation_errors = []
+
+        for _, field in self._config.items():
+            if not self.dependencies_satisfied(field):
+                # we don't validate a field if its dependencies are not met
+                logger.info(
```

Comment:
shouldn't this be `debug` or `warning` ?

Reply:
It's not an issue if there are no dependencies, so I'll change it to `debug`.